internal/docker/podman.go

```diff
@@ -0,0 +1,236 @@
+
+	localRootlessPath     = getLocalRootlesPath()
+	defaultRemoteRootless = fmt.Sprintf("ssh://core@localhost:63753%s?secure=false", localRootlessPath)
+	defaultLocalRootless  = fmt.Sprintf("unix://%s", localRootlessPath)
+)
+
+func getLocalRootlesPath() string {
+	return fmt.Sprintf("unix:///run/user/%d/podman/podman.sock", os.Getuid())
+}
+
+func podmanOverSSH() (*client.Client, error) {
+	log.Debug("using docker client to connect to podman over ssh")
+	var clientOpts = []client.Opt{
+		client.WithAPIVersionNegotiation(),
+		// TODO: is this needed?
+		client.FromEnv,
+	}
+
+	host := defaultRemoteRootless
+	makeClient := sshClient
+
+	if v, found := os.LookupEnv("CONTAINER_HOST"); found && v != "" {
+		log.Debugf("using $CONTAINER_HOST: %s", v)
+		host = v
+	}
+
+	if v, found := os.LookupEnv("PODMAN_HOST"); found && v != "" {
+		log.Debugf("using $PODMAN_HOST: %s", v)
+		host = v
+	}
+
+	log.Debugf("host: %q", host)
+
+	hostURL, err := url.Parse(host)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing host %s with: %w", host, err)
+	}
+
+	httpClient, err := makeClient(hostURL)
+	if err != nil {
+		return nil, fmt.Errorf("making http client: %w", err)
+	}
+
+	clientOpts = append(clientOpts, func(c *client.Client) error {
+		return client.WithHTTPClient(httpClient)(c)
+	})
+	clientOpts = append(clientOpts, client.WithHost("http://d"))
+
+	dockerClient, err := client.NewClientWithOpts(clientOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed create remote client for podman: %w", err)
+	}
+
+	return dockerClient, err
+}
+
+func podmanViaUnixSocket() (*client.Client, error) {
+	log.Debug("using docker client to connect to podman over local unix socket")
+	var clientOpts = []client.Opt{
+		client.WithAPIVersionNegotiation(),
+		// TODO: is this needed?
+		client.FromEnv,
+	}
+
+	// TODO: note this is shared code
+	//if v, found := os.LookupEnv("PODMAN_HOST"); found && v != "" {
+	//	log.Debugf("using $PODMAN_HOST: %s", v)
+	//	host = v
+	//}
+
+	// TODO: if DOCKER_HOST is configued will this take precidence?
+	clientOpts = append(clientOpts, client.WithHost(getLocalRootlesPath()))
+
+	dockerClient, err := client.NewClientWithOpts(clientOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed create local client for podman: %w", err)
+	}
+
+	return dockerClient, err
+}
+
+func GetClientForPodman() (*client.Client, error) {
+	log.Debug("creating podman client")
+
+	// TODO: how should detection work here?
+	switch runtime.GOOS {
+	case "windows", "darwin":
+		return podmanOverSSH()
+	default:
+		return podmanViaUnixSocket()
+	}
+}
+
+type clientMaker func(*url.URL) (*http.Client, error)
+
+func unixClient(hostURL *url.URL) (*http.Client, error) {
+	return &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				return (&net.Dialer{}).DialContext(ctx, "unix", hostURL.Path)
+			},
+			DisableCompression: true,
+		},
+	}, nil
+}
```

Comment:
These could be deleted now (I think it was me that didn't clean this up while we were troubleshooting)